activestorage/lib/active_storage/attached/many.rb

```diff
@@ -44,20 +44,25 @@ def detach
       attachments.destroy_all if attached?
     end
 
+    ##
+    # :method purge
+    #
+    # :call-seq: User.highlights.purge
+    #
     # Directly purges each associated attachment (i.e. destroys the blobs and
     # attachments and deletes the files on the service).
-    def purge
-      if attached?
-        attachments.each(&:purge)
-        attachments.reload
-      end
-    end
+    #
+    # Attachments can be selectively purged using queries
+    # e.g. User.highlights.where(id: []).purge
 
-    # Purges each associated attachment through the queuing system.
-    def purge_later
-      if attached?
-        attachments.each(&:purge_later)
-      end
-    end
+    ##
+    # :method purge_later
+    #
+    # :call-seq: User.highlights.purge_later
+    #
+    # Purges each associated attachment through the queing system
+    #
+    # Attachments can be selectively purged later using queries
+    # e.g. User.highlights.where(id: []).purge_later
```

Comment:
I'm not even sure this extra documentation makes sense. We're making it so that users can further scope the `user.highlights` query — like they can with any other Active Record association.

Reply:
Will remove in next push